README.md

```diff
@@ -4,6 +4,66 @@
 
 Documentation available at [http://www.oskari.org].
 
+This repository holds the front-end code for Oskari. Developing the front-end requires an oskari-server to be running that responds to the XHR requests made by the front-end. You can download a pre-compiled copy of the server from  [http://www.oskari.org/download] or checkout the source code in the oskari-server repository. Any customized application should use the oskari-server-extension template as base for customized server. The front-end code is build using Webpack.
+
+## Preparations
+
+Make sure you have at least Node 8.x / NPM 5.x. Run `npm install` in the front-end repository root.
+Make sure you have oskari-server running on localhost port 8080 (can be customized on webpack.config.js).
+
+## Run in development
+
+Webpack dev server is used to serve the JS bundle and assets when running in local development. XHR calls will be proxied to the Java backend assumed to be running on localhost:8080.
+
+So that the server knows to look for the JS bundle and assets from the right place, we need to have the client version configured in the Oskari-server `oskari-ext.properties`:
+
+```
+oskari.client.version=dist/devapp
+```
+
+To start Webpack dev server, we point it to an application directory with subfolders having miniferAppSetup.json files for the applications we want to start, here Sample app as example:
+`npm start -- --env.appdef=applications/sample/`
+
+When you see "Compiled successfully." in the terminal, you can open the app in the browser at `localhost:8081`.
+
+The dev server has automatic reloading enabled when you save changes to JS code and hot reloading for S/CSS without need for full browser reload.
+
+## Build for production
+
+To build minifed JS and assets, run:
+`npm run build -- --env.appdef=1.49.0:applications/sample/`
+
+The number before the colon sets the directory name, here producing files under dist/1.49.0/servlet/
+
+Note: The version number given for the build command needs to match the client version (`oskari.client.version`) in Oskari-server `oskari-ext.properties`.
+
+### Customized application icons
+
+For customizing icons include a folder named `icons` under the application folder (for example in sample app `applications/sample/servlet/icons`) with files named similarly (and with same pixel size for maximum compatibility) as in `oskari-frontend/resources/icons`. You can do a partial customization by for example just having icon-arrow-down.png in the application icons folder and the build will create an icon-set with default icons only overriding the icon-arrow-down.png.
```

Comment:
Maybe we should start this section with emphasising that this step is only needed if the application actually has custom icons?

Also the last sentence is somewhat long and hard to parse. It could be misunderstood that the default icons are overriding, not the other way around.